src/tests/Microsoft.PowerFx.Core.Tests/CheckResultTests.cs

```diff
@@ -479,6 +447,36 @@ public void TestApplyGetLogging(string expr, string execptedLog, bool success)
             var log = check.ApplyGetLogging();
             Assert.Equal(success, check.IsSuccess);
             Assert.Equal(execptedLog, log);
+        }
+
+        private void CheckResultExpectedReturnValue(string inputExpr, bool allowCoerceTo, bool isSuccess, string errorMsg, FormulaType returnType)
+        {
+            var check = new CheckResult(new Engine())
+                .SetText(inputExpr)
+                .SetBindingInfo()
+                .SetExpectedReturnValue(returnType, allowCoerceTo);
+
+            if (isSuccess)
+            {
+                Assert.True(check.IsSuccess);
+            }
+            else
+            {
+                string exMsg = null;
+
+                try
+                {
+                    var errors = check.ApplyErrors();
+                    exMsg = errorMsg.ToString();
+                    Assert.False(check.IsSuccess);
+                }
+                catch (Exception ex)
```

Comment:
>catch (Exception ex) [](http://example.com/codeflow?start=16&length=20)

Who is throwing? Is it check.ApplyErrors()?

Reply:
Yes, check.Applyerrors will throw exception of not implementation method if expectedreturntype is not string or number